### PR TITLE
Refactor Popup

### PR DESCRIPTION
**Pull request by mariusraht2**
Related to #3040

Refactored logic and call of popups to be able to be reused for showing and selecting available commits.

### CODE REVIEW

**Comment by larshp** [comment]
good size PR, thanks 👍

**Comment by mariusraht2** [comment]
Thanks, I hope I can keep it up^^ :)

File: src/ui/zcl_abapgit_services_git.clas.abap [hunk 1]
```diff
@@ -307,8 +313,11 @@ CLASS zcl_abapgit_services_git IMPLEMENTATION.
 
         zcl_abapgit_objects=>delete( it_tadir  = lt_selected
                                      is_checks = ls_checks ).
-* update repo cache
-        lo_repo->refresh( ).
+
+        IF iv_refresh = abap_true.
```
Comment (larshp) [inline src/ui/zcl_abapgit_services_git.clas.abap:317]: what does this do? is it part of refactoring the popup functionality?

`IF iv_refresh = abap_true.`

Comment (mariusraht2) [reply src/ui/zcl_abapgit_services_git.clas.abap:317]: No, it doesn't directly belong to this PR - will remove it. Later via iv_refresh it's possible to control wether to refresh the repo or not to be able to suppress a unneccessary repo refresh.

**Pull request merged**